docs/en/middleware/zenoh.md

```diff
@@ -0,0 +1,194 @@
+
+![Architecture PX4 Zenoh-Pico with ROS 2](../../assets/middleware/zenoh/architecture-px4-zenoh.svg)
+
+The client is the _PX4 Zenoh-Pico Node_ referred to above, which is implemented in the [PX4 `zenoh` module](../modules/modules_driver.md#zenoh).
+This is based on Zenoh-Pico, a minimalistic version of [Eclipse Zenoh](https://zenoh.io/) (a data-centric protocol designed for real-time, distributed, and resource-constrained environments).
+
+The router suggested above is [zenohd](https://github.com/eclipse-zenoh/zenoh/tree/main/zenohd).
+
+:::info
+UART is supported by Zenoh but has not yet implemented in the PX4 Zenoh-Pico node.
+:::
+
+## ROS 2 Zenoh Bring-up on Linux Companion
+
+In order for PX4 uORB topics to be shared with ROS 2 applications, you will need the PX4 Zenoh-Pico Node client running on your FMU, connected to a Zenoh router running on the companion computer (or elsewhere in the network).
+
+First select Zenoh as the ROS 2 transport by setting the `RMW_IMPLEMENTATION` environment variable as shown:
+
+```sh
+export RMW_IMPLEMENTATION=rmw_zenoh_cpp
+```
+
+Then start the Zenoh router using the command:
+
+```sh
+ros2 run rmw_zenoh_cpp rmw_zenohd
+```
+
+For more information about the Zenoh Router see the [rmw_zenoh](https://github.com/ros2/rmw_zenoh?tab=readme-ov-file#start-the-zenoh-router) documentation.
+
+## PX4 Zenoh-Pico Node Setup
+
+### PX4 Firmware
+
+Before setting up the Zenoh communication, first make sure that your firmware contains the driver that implements the [`zenoh` driver](../modules/modules_driver.html#zenoh), which provides the implementation of the _PX4 Zenoh-Pico Node_.
+
+You can check if the module is present on your board by searching for the key `CONFIG_MODULES_ZENOH=y` in your board's `default.px4board` KConfig file.
+For example, you can see that the module is present in `px4_fmu-v6xrt` build targets from [/boards/px4/fmu-v6xrt/default.px4board](https://github.com/PX4/PX4-Autopilot/blob/main/boards/px4/fmu-v6xrt/default.px4board#L91).
+If it is not present you can add this key to your board configuration and rebuild.
+
+The table below shows some of the PX4 targets that include this module (at time of writing).
+
+| PX4 Target             | Notes                                                                                       |
+| ---------------------- | ------------------------------------------------------------------------------------------- |
+| `px4_fmu-v6xrt`        | For [FMUv6X-RT](../flight_controller/nxp_mr_vmu_rt1176.md) (reference platform for testing) |
+| `nxp_tropic-community` |                                                                                             |
+| `nxp_mr-tropic`        |                                                                                             |
+| `nxp_mr-canhubk344`    |                                                                                             |
```

Comment:
I don't know what these are for. If that is useful perhaps add note (it might be something that NXP users will already know about).